pkg/core/mem_pool.go

```diff
@@ -0,0 +1,250 @@
+func NewMemPool(capacity int) MemPool {
+	return MemPool{
+		unsortedTxn:   make(map[util.Uint256]*PoolItem),
+		unverifiedTxn: make(map[util.Uint256]*PoolItem),
+		capacity:      capacity,
+	}
+}
+
+// TryGetValue returns a transactions if it esists in the memory pool.
+func (mp MemPool) TryGetValue(hash util.Uint256) (*transaction.Transaction, bool) {
+	if pItem, ok := mp.unsortedTxn[hash]; ok {
+		return pItem.txn, ok
+	}
+
+	if pItem, ok := mp.unverifiedTxn[hash]; ok {
+		return pItem.txn, ok
+	}
+
+	return nil, false
+}
+
+// getLowestFeeTransaction returns the PoolItem with the lowest fee amongst the "verifiedTxnSorted"
+// and "unverifiedTxnSorted" PoolItems along with a integer. The integer can assume two values, 1 and 2 which indicate
+// that the PoolItem with the lowest fee was found in "verifiedTxnSorted" respectively in "unverifiedTxnSorted".
+// "verifiedTxnSorted" and "unverifiedTxnSorted" are sorted slice order by transaction fee ascending. This means that
+// the transaction with lowest fee start at index 0.
+// Reference: GetLowestFeeTransaction method in C# (https://github.com/neo-project/neo/blob/master/neo/Ledger/MemoryPool.cs)
+func getLowestFeeTransaction(verifiedTxnSorted PoolItems, unverifiedTxnSorted PoolItems) (*PoolItem, int) {
+	minItem := min(unverifiedTxnSorted)
+	verifiedMin := min(verifiedTxnSorted)
+	if verifiedMin == nil || (minItem != nil && verifiedMin.CompareTo(minItem) >= 0) {
+		return minItem, 2
+	}
+
+	minItem = verifiedMin
+	return minItem, 1
+
+}
+
+// min return the minimum item in a ascending sorted slice of pool items.
+// The  function can't be applied to unsorted slice!
+func min(sortedPool PoolItems) *PoolItem {
+	var minItem *PoolItem
+	if len(sortedPool) > 0 {
+		minItem = sortedPool[0]
+	} else {
+		minItem = nil
+	}
+	return minItem
+}
+
+// GetVerifiedTransactions returns a slice of Input from all the transactions in the memory pool
+// whose hash is not included in excludedHashes.
+func (mp *MemPool) GetVerifiedTransactions() []*transaction.Transaction {
+	var t []*transaction.Transaction
+	for _, p := range mp.unsortedTxn {
+		t = append(t, p.txn)
+	}
+
+	return t
+}
+
+// Verify verifies if the inputs of a transaction tx are already used in any other transaction in the memory pool.
+// If yes, the transaction tx is not a valid transaction and the function return false.
+// If no, the transaction tx is a valid transaction and the function return true.
+func (mp MemPool) Verify(tx *transaction.Transaction) bool {
+	var mpInputs []transaction.Input
+
+	mpTxn := mp.GetVerifiedTransactions()
+	for _, t := range mpTxn {
+		if t.Hash() != tx.Hash() {
+			for _, in := range t.Inputs {
+				mpInputs = append(mpInputs, *in)
+			}
+		}
+	}
+
+	var txInputs []transaction.Input
+	for _, in := range tx.Inputs {
+		txInputs = append(txInputs, *in)
+	}
+
+	if i := transaction.InputIntersection(mpInputs, txInputs); len(i) > 0 {
+		return false
+	}
+	return true
+}
```

Comment:
is that more simple?
```go
func (mp MemPool) Verify(tx *transaction.Transaction) bool {
	count := 0
	inputs := make([]*transaction.Input, 0)
	for _, item := range mp.GetVerifiedTransactions() {
		if tx.Hash().Equals(item.Hash()) {
			inputs = append(inputs, item.Inputs...)
		}
	}
	
	for i := 0; i < len(inputs); i++ {
		for j := 0; j < len(tx.Inputs); j++ {
			if inputs[i].PrevHash.Equals(tx.Inputs[j].PrevHash) {
				count++
			}
		}
	}

	return count == 0
}
```

Reply:
:+1: 
By using your proposed solution, I don't need anymore the `InputIntersection` method in the input.go file. So we can further simplify the code.